jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -352,6 +352,29 @@ public static String toJarPath(String jarFile, String pathInJar)
         return "jar:" + jarFile + URIUtil.addPaths("!/", pathInJar);
     }
 
+    /**
+     * Unwrap a URI to expose its container path reference.
+     *
+     * Take out the container archive name URI from a {@code jar:file:${container-name}!/} URI.
+     *
+     * @param uri the input URI
+     * @return the container String if a {@code jar} scheme, or just the URI untouched.
+     */
+    public static URI unwrapContainer(URI uri)
```

Comment:
Since that's completely `Resource`-agnostic, this can probably be moved to `URIUtil`.

Reply:
Absolutely, and is one of several static methods that will move with that future FileID/URIUtil PR.